include/onnxruntime/core/framework/op_kernel_context.h

```diff
@@ -74,7 +74,7 @@ class OpKernelContext {
   // and shape must specify the shape of the underlying dense-tensor.
   // Memory allocation for the output may happen when this method is invoked,
   // unless static optimization pre-allocates it.
-  SparseTensor* Output(int index, size_t num_values, const TensorShape& shape);
+  SparseTensor* OutputSparse(int index, const TensorShape& shape);
```

Comment:
What is the model for memory-allocation for sparse-tensors? The ```num_values``` parameter helps determine the allocation-size. How does it work if this parameter is not specified?

Reply:
This worked for COO format where nnz deermined the size of all of the allocations values and indices. But this is no longer the case. In this PR we simply allocated an instance with the supplied allocator. In the next PRs I will have to come up with the strategy to cover different scenarious including allocation re-use whenether possible.

Reply:
memory allocation and memory planning is in the next PR

Reply:
Okay. I still think that some mechanism is needed for the kernel to indicate how much memory to allocate, since only the kernel knows this.

Reply:
I see, so the original call doesn't really allocate the memory, it is done later. That's fine then, thanks.